### ISSUE TITLE
Add Syncthing 0.11.x

### ISSUE THREAD

**Issue by TimVNL** [SynoCommunity/spksrc#1674]
Syncthing has a new major release with lots of improvements and fixes see:

https://github.com/syncthing/syncthing/releases/tag/v0.11.0
https://github.com/syncthing/syncthing/releases/tag/v0.11.1
https://github.com/syncthing/syncthing/releases/tag/v0.11.2

Would love to see it in the repo

**Comment by Lorefyr** [comment]
I'd be very happy about this as well.

**Comment by ArthurvdMolen** [comment]
Me, too in my package center Syncthing v0.10.30-3 beta does not show up. Any clue? Iss on the home page...

**Comment by Moter8** [comment]
:+1:

**Comment by voltige** [comment]
With current syncthing 0.11.5, synchonisation no longer performs with older versions. By cons, it works perfectly between current versions.

Please update it. 

I also love a full and accurate tutorial to go from source code to a .spk

**Comment by ptxmac** [comment]
:+1:

**Comment by struthio** [comment]
Also would like update to 0.11 version :)

**Comment by LucaTony** [comment]
Please update to 0.11 ... otherwise I can't use my server

**Comment by phishlite** [comment]
beware :  release 0.11 contains protocol changes. It will not connect to v0.10.x clients.
Temporary solution (until v 0.11 spk...) : downgrade your clients to v0.10.30  https://github.com/syncthing/syncthing/releases/tag/v0.10.30
(and don't forget to  deactivate auto update function)

**Comment by Zetten** [comment]
For what it's worth, the upgrade seems to have been perfectly smooth for me by simply changing `spk/syncthing/Makefile` and `cross/syncthing/Makefile` to reflect the new version number (0.11.8). The actual compilation went off almost without a hitch - I needed a full clean of the go build path; I think some 0.10.x binaries were getting in the way.

After that, the installation on my NAS (DS414, armadaxp) was fine and it seems to be happily syncing away right now.

From a completely clean spksrc clone:
```
vim {spk,cross}/syncthing/Makefile # update the syncthing package version
make setup
cd native/go
make
cd ../../spk/syncthing
make arch-$YOUR_NAS_ARCH
```

I'd be happy to commit the change and submit a PR, if anyone else can confirm it's working, perhaps on different architectures.

**Comment by DaftHonk** [comment]
I also built a spk for my 1813+ in the same manner, but after running the package for a short time Package Center says the package is stopped but the Web interface is still accessible. A minor inconvenience but will need to track down the cause nonetheless. May be related to auto updates.

**Comment by TimVNL** [comment]
Can't we make a new spk with the name syncthing-0.11 so both versions are present in Package Center?

**Comment by Zetten** [comment]
@DaftHonk My package has been in the 'running' state happily since I posted originally (less an internal restart or two for config changes), but I'll definitely keep an eye on it. How long did it take for the issue to present itself?

@TimVNL That makes some sense given the protocol incompatibility, but the main issue I can see would be handling the upgrade from 0.10.x to 0.11.x. The file index does need to be recreated, but the host ID, remembered peers, and synced folders are all retained if you upgrade the existing package. Maybe that's not an issue for some, but I reckon losing that and having to reconfigure would be a big downside for users with a lot of peers or folders.

The two versions will conflict with listening ports and so on if no other changes are made, so the best thing to do from a package management perspective would be to mark them as conflicting (or 'provides' or whatever the equivalent is). But my guess is that the upgrade/replace mechanism in question would totally uninstall syncthing before installing syncthing-0.11, and even if not they would end up in different target directories (which seem to be based on package names).

I don't know much about the spk system - is it possible to handle this kind of situation? A brute force kind of workaround would be moving all the data out of `/var/packages/syncthing/target` and into an external folder which is kept safe through a package removal/reinstall, but that also then requires an upgrade to the syncthing 0.10.x package to externalise this data in the first place.

**Comment by Dr-Bean** [comment]
Simply put: Until the Syncthing devs say that 10.x cannot be upgraded to 11.x, there's no need to consider other options. 
When someone (anyone) gets around to opening a PR with the correct changes, tested and proven: great, that would be appreciated.

**Comment by Zetten** [comment]
There it is, but as mentioned it would be great for some additional feedback on the new package from other testers. I haven't noticed the 'running-but-not' problem myself yet - it all looks good to me.

**Comment by DaftHonk** [comment]
I created my initial package with release 0.11.6 so I believe the 'running-but-not' issue occurs when the program auto-updates. It hasn't occurred again since it updated to 0.11.8.

**Comment by Zetten** [comment]
Ah, I see. I do vaguely recall a similar thing happening to me after an auto-update on the 0.10.x line as well, so it may not be related to the 0.11.x upgrade. That was ages ago so I could be mistaken though.

Does the auto-update mechanism also change the package metadata reported through the Package Center?

**Comment by TimVNL** [comment]
Made a simple SPK repo untill official comes out.... see http://spk,timveluwenkamp.eu

**Comment by NickPyz** [comment]
I'd like to shed some light on this:

The latest Synocommunity release is v0.10.30-3. The lead developer at Syncthing (Jakob Borg) stated to me that v0.10.30 was broken - the version upgrade option was missing. Here is a link to the ST user forum thread:

https://forum.syncthing.net/t/synology-ds412-update-to-latest-version/3024/2

Synology users downloading the current Synocommunity version will encounter 2 problems:

(1) They can't upgrade to the newer v0.11.x series. They are stuck.

(2) Compounding the problem, devices running v0.11x will not sync with devices running v0.10.x or earlier.

The only solution appears to be to compile and release a new Syno version based on a more recent   executable .... ST is presently at v0.11.9

Thank you

**Comment by Zetten** [comment]
@DaftHonk I've seen the 'running-but-not' issue now on my own system after syncthing autoupdated to 0.11.9. The core issue seems to be with the pidfile (`/usr/local/syncthing/var/syncthing.pid`) managed by start-stop-daemon not getting updated. It simply points to the old process id from before the autoupdate.

I don't think it's strictly related to the 0.11.x upgrade, but I'm not sure how best to go about fixing it.

Presumably this is a problem affecting anyone who's written a syncthing initscript on other *nix systems using start-stop-daemon though, so perhaps there is inspiration somewhere out there...

**Comment by cytec** [comment]
@Zetten you could just do a `ps | grep /usr/local/syncthing/bin/syncthing` for status and use `awk '{print $1}'` on that to get the pid for stopping i think some packages used that in some previous version...

not really clean and stuff but _should_ work... haven't tested it...

**Comment by Zetten** [comment]
@cytec Going by the (aborted) PR syncthing/syncthing#1277 that should work (using `pidof` rather than `ps | grep`).

We should just bear in mind that syncthing actually launches several processes, a manager and a worker in my case (maybe more workers in some circumstances?), so in the case of stopping we need to kill all of the pids, or else the manager will just restart any killed workers. Like you say - it's not clean, but it should be functional.

Disabling autoupgrade was my backup plan - that also solves the problem of the actual version number diverging from that reported in Package Center. But it's not ideal for other reasons, not least that someone will have to frequently update the spk Makefile to keep up with the regular syncthing releases.

**Comment by cytec** [comment]
if i remember correctly release cycles should slow down once it hits 1.x... not 100% sure but i think i've read something like that some while ago

**Comment by Kaelber** [comment]
I´m using Synology Diskstations DS1512+, 412+, 712+ and some others. This three are x86-models and currently only in the DS712+ model I can choose the installation of Syncthing. In the past, also the two other models are able to install Syncthing over the package center. I don´t know exactly, but it seems since DSM 5.2, is installed in all my machines, there are differences

What can I doe? I´m also search the SPK´s for Syncthing for a manuall installation, but I doesn´t find. Is possible to get a Download adress for this?

**Comment by Zetten** [comment]
@cytec It's actually an even simpler fix than expected - removing the pidfile stuff (`-p` and `-m` args) from start-stop-daemon reverts to process matching by owner/executable. This works perfectly well for detecting and stopping the multiple syncthing processes in my quick tests.

I've added this as a second commit on the PR.

I think it's easiest to leave the current behaviour of allowing auto-updating for now. If as you say the syncthing releases slow down, disabling automatic upgrades could be revisited then.

**Comment by chrisbramley** [comment]
Would be spiffing if this was available. I'm not super-clued on linux, and whilst I have learned a lot using the Synology at cmd line I'm very keen for this to be a package upgrade as all the others in the community are. 

0.11.x seems to have some helpful changes. Can anyone post an etr for the new version in spk?

**Comment by symera** [comment]
I just found a solution without the need of compiling or having to wait for an updated spk. Works out for me so far. Recently updated to v0.11.10 and it's syncing 4 devices without any issues. Here is what I have done:
- installed the package (v0.10.30-4 beta) by adding the synocommunity package source
- replaced the binary (./syncthing) at /var/packages/syncthing/target/bin/ with the most recent* one from https://github.com/syncthing/syncthing/releases/download/v0.11.9/syncthing-linux-arm-v0.11.9.tar.gz
* I did it yesterday, so the most recent version has been v0.11.9. So I didn't tried it with the new release (v0.11.10). The update from Syncthing itself did the job now, but I guess it'll be good with the new one as well.

**Comment by Zetten** [comment]
Yeah, that's pretty much all the automatic upgrade does (download and replace the binary). The only thing to watch out for is that you get the right CPU architecture - your link is fine for any ARM models but won't work on an x86 NAS.

I can also confirm the revised start/stop/status script in my PR still works perfectly after an automatic upgrade. So, as far as I think is reasonable, the issue will be resolved once that is merged and the next builds roll out to the SynoCommunity repos.

**Comment by symera** [comment]
Yes, of course, you've to choose the right binary for your arch. Thanks for pointing that out.

**Comment by chrisbramley** [comment]
Seems to work so far with 0.11.10!

**Comment by matthias-otto** [comment]
After adding http://packages.synocommunity.com/ as a package source, I see many new packages, but I do not see a Syncthing package for installation. Has it been removed because of the problems discussed here?

Obviously, with no package to install, doing any fixes involving swapping out binaries becomes measurably harder :)

**Comment by chrisbramley** [comment]
Have you allowed viewing of Betas in settings?  


     On Tuesday, June 23, 2015 10:22 AM, matthias-otto <notifications@github.com> wrote:
   

 After adding http://packages.synocommunity.com/ as a package source, I see many new packages, but I do not see a Syncthing package for installation. Has it been removed because of the problems discussed here?Obviously, with no package to install, doing any fixes involving swapping out binaries becomes measurably harder :)—

**Comment by matthias-otto** [comment]
Doh. Thank you for solving my problem :)

**Comment by chrisbramley** [comment]
:) 

So far, this version seems to be scanning much more quickly. Looking good. 
 


     On Tuesday, June 23, 2015 11:23 AM, matthias-otto <notifications@github.com> wrote:
   

 Doh. Thank you for solving my problem :)—

**Comment by singpolyma** [comment]
I have enabled beta versions, but do not see a syncthing option in my package centre after adding this source either...

**Comment by matthias-otto** [comment]
For me, activating beta versions solved it. The only other thing that
stumped me for a bit is that the rider "all" does not show any community
software. Only "Community" will list stiff like Syncthing.

-- 
incub8 Software Labs GmbH
Pulverstraße 1 | 44225 Dortmund, Germany
+49 (0) <+49%20(0)231%2077656397>231 77656397 <+49(0)23177656397> |
info@incub8.de | http://www.incub8.de/

Geschäftsführer: Jan Dörrenhaus | Sitz: Dortmund
Registergericht: Amtsgericht Dortmund - HRB 25457
Steuernummer: 315/5748/1377 | USt.-Id.: DE287331286

Bankverbindung: Sparkasse Dortmund
Kto.-Nr.: 911008424 | BLZ: 440 501 99
IBAN: DE64440501990911008424 | BIC: DORTDE33


2015-06-24 3:34 GMT+02:00 Stephen Paul Weber <notifications@github.com>:

**Comment by jb510** [comment]
Getting closer...

So I installed the 0.10.30 package via synocommunity on my DS213 (ArmV5te) and it wouldn't start, just silently failed.  `ps | grep /usr/local/syncthing/bin/syncthing` actually showed it running, but I think maybe that's only part of it.

Tried using `wget` to pull down ../syncthing-linux-arm-v0.11.10.tar.gz untar it and replace the binary manually, but that wouldn't start either.  Aparently there is a problem with Go on Armv5: https://github.com/SynoCommunity/spksrc/pull/1631#issuecomment-94900688 :(

So I finally went back an did `wget https://github.com/syncthing/syncthing/releases/download/v0.10.30/syncthing-linux-arm-v0.10.30.tar.gz` and copied THAT binary over what was now the now working 0.11.10 binary.. and THAT amazingly is working.

Unfortunately all my other clients are 0.11.10 so no syncing, but at least got the GUI up.

This post is to raise the issue about ArmV5 support or lack their of in the 0.11.10 build.  

Any further insight that would get me up in running on 0.11.x would be welcome, but I'd need a lot of step by step directions to build syncthing from source if that's a viable fix.

**Comment by jb510** [comment]
Workaround:
1. I installed Syncthing 10.30 from the synocommunity repo.
2. I replaced the Syncthing 10.30 binary with the 10.30 binary from the [syncthing-linux-arm-v0.10.30] 
(you might be able to skip this and go straight to 11.10.
3. run Syncthing 10.30
4. Replace the SyncThing 10.30 binary with the 11.10 binary.
5. Run SyncThing from the DSM GUI (it'll still errantly report the version as 10.30-3, but it's really 11.10)

To replace the binaries:
a) ssh into your syno box as root or syncthing 
b) cd /volume1/@appstore/syncthing/bin
c) mkdir temp
d) cd temp
e) wget https://github.com/syncthing/syncthing/releases/download/v0.10.30/syncthing-linux-arm-v0.10.30.tar.gz (you might be able to skip this and go straight to 11.10, I didn't try)
f) wget https://github.com/syncthing/syncthing/releases/download/v0.11.10/syncthing-linux-arm-v0.11.10.tar.gz
g) tar -xzvf syncthing-linux-arm-v0.10.30.tar.gz
e) cp syncthing-linux-arm-v0.10.30/syncthing ../. 
f) go to the diskstation GUI and run syncthing, confirm it's working, then stop syncthing, return to ssh
g) tar -xzvf syncthing-linux-arm-v0.11.10.tar.gz
f) cp syncthing-linux-arm-v0.11.10/syncthing ../. 
g) go to the diskstation GUI and run syncthing, confirm it's working, it told me I had to restart it to update settings.
h) enjoy
<insert long list of disclaimers>

The main point being 11.10 does run fine on my DS213 (arm5), the consolidated the various syncthing builds into a single arm build, and that includes arm5.

**Comment by newkind** [comment]
I did a manual update of the syncthing binary, but the package stops by itself from time to time - not sure why this happens ... Updated package in the Synocommunity repo would be ideal.

**Comment by symera** [comment]
The repo is still deprecated unfortunately, so just fyi: If I launch "Package Center" in the DSM to check the status of Syncthing it seems to be stopped. But nevertheless Syncthing is running fine, I just checked the webgui (http://device:7070). Maybe the "old" spk is not checking the status of the following versions (which I replaced once to keep it going). But I don't mind, since the auto update works and the syncing itself works flawlessly so far.
